dandiapi/api/views/dandiset.py

```diff
@@ -82,9 +82,34 @@ def get_queryset(self):
         # TODO: This will filter the dandisets list if there is a query parameter user=me.
         # This is not a great solution but it is needed for the My Dandisets page.
         queryset = Dandiset.objects.all().order_by('created')
-        user_kwarg = self.request.query_params.get('user', None)
-        if user_kwarg == 'me':
-            return get_objects_for_user(self.request.user, 'owner', queryset, with_superuser=False)
+        if self.action == 'list':
+            user_kwarg = self.request.query_params.get('user', None)
+            if user_kwarg == 'me':
+                queryset = get_objects_for_user(
+                    self.request.user, 'owner', queryset, with_superuser=False
+                )
+            # Same deal for filtering out draft or empty dandisets
+            show_draft = self.request.query_params.get('draft', 'false') == 'true'
+            show_empty = self.request.query_params.get('empty', 'false') == 'true'
+
+            if not show_draft:
+                # Only include dandisets that have more than one version, i.e. published dandisets.
+                queryset = queryset.annotate(version_count=Count('versions')).filter(
+                    version_count__gt=1
+                )
+            if not show_empty:
+                # Only include dandisets that have assets.
+                # It is enough to check the draft version, since any published version is required
```

Comment:
Wouldn't this exclude a dandiset with a draft version that has no assets, but a published version that does?

In other words, if the user uploaded assets to a dandiset, published it, then deleted them from the draft version, that dandiset would be regarded as an "empty" dandiset here.

Reply:
Good point